src/vs/workbench/api/node/extHostTerminalService.ts

```diff
@@ -41,7 +41,7 @@ export class ExtHostTerminalService extends BaseExtHostTerminalService {
 		// Getting the SystemShell is an async operation, however, the ExtHost terminal service is mostly synchronous
 		// and the API `vscode.env.shell` is also synchronous. The default shell _should_ be set when extensions are
 		// starting up but if not, we run getSystemShellSync below which gets a sane default.
-		getSystemShell(platform.platform).then(s => this._defaultShell = s);
+		getSystemShell(platform.platform, process.env as platform.IProcessEnvironment).then(s => this._defaultShell = s);
```

Comment:
I can still use these here because it's in the extension host service?

Reply:
Yes, extension host is safe.